doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -84,37 +175,265 @@ writeTextFile {
   executable = true;
   destination = "/bin/my-file";
 }
-# Writes contents of file to /nix/store/<store path>
+```
+:::
+
+::: {.note}
+The commands `writeText`, `writeTextDir`, `writeScript`, `writeScriptBin`, `writeShellScript`, and `writeShellScriptBin` documented below are convenience functions that wrap `writeTextFile`.
+:::
+
+### `writeText` {#trivial-builder-writeText}
+
+Writes a text file to the store
+
+`writeText` takes two arguments: `name` and `text`, each of which should be
+a string.
+
+`name`
+
+: the name used in the Nix store path.
+
+`text`
+
+: will be the contents of the file.
+
+The store path will include the the name, and it will be a file. Any path separators and shell-reserved elements in the name are escaped to create the store path identifier.
+
+Here is an example.
+
+::: {.example #ex-writeText}
+# Usage of `writeText`
+```nix
+# Writes my-file to /nix/store/<store path>
 writeText "my-file"
   ''
   Contents of File
   '';
+```
+:::
+
+This example is a simpler way to spell:
+
+```nix
+writeTextFile {
+  name = "my-file";
+  text = ''
+    Contents of File
+  '';
+}
+```
+
+### `writeTextDir` {#trivial-builder-writeTextDir}
+
+Writes a text file within a subdirectory of the store.
+
+`writeTextDir` takes two arguments: `path` and `text`, each of which should be a string.
+
+`path`
+
+: the destination within the Nix store path under which to create the file.
+
+`text`
+
+: the contents of the file.
+
+The store path will be a directory. The Nix store identifier will be generated based on various elements of the path.
```

Comment:
Is the second sentence here necessary? "The Nix store identifier will be generated based on various elements of the path."